recipes/libqasm/all/conanfile.py

```diff
@@ -0,0 +1,113 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMakeToolchain, CMakeDeps, CMake, cmake_layout
+from conan.tools.files import get
+from conan.tools.microsoft import check_min_vs, is_msvc
+from conan.tools.scm import Version
+import os
+
+class LibqasmConan(ConanFile):
+    name = "libqasm"
+
+    # Optional metadata
+    license = "Apache-2.0"
+    homepage = "https://github.com/QuTech-Delft/libqasm"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Library to parse cQASM files"
+    topics = ("code generation", "parser", "compiler", "quantum compilation", "quantum simulation")
+
+    # Binary configuration
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "build_python": [True, False],
+        "build_tests": [True, False],
+        "compat": [True, False],
+        "tree_gen_build_tests": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "build_python": False,
+        "build_tests": False,
+        "compat": False,
+        "tree_gen_build_tests": False
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 23
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "apple-clang": "13",
+            "clang": "13",
+            "gcc": "11",
+            "msvc": "19"
+        }
+
+    def build_requirements(self):
+        self.build_requires("cmake/3.20.1")
+        self.tool_requires("m4/1.4.19")
+        if self.settings.os == "Windows":
+            self.tool_requires("winflexbison/2.5.24")
+        else:
+            self.tool_requires("flex/2.6.4")
+            self.tool_requires("bison/3.8.2")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def layout(self):
+        self.folders.source = "."
+        self.folders.build = os.path.join("build", str(self.settings.build_type))
+        self.folders.generators = os.path.join(self.folders.build, "generators")
+
+        self.cpp.package.libs = ["cqasm"]
+        self.cpp.package.includedirs = ["include"]
+        self.cpp.package.libdirs = ["lib"]
+
+        self.cpp.source.includedirs = ["include"]
+        self.cpp.build.libdirs = ["."]
```

Comment:
Isn't this mostly what `cmake_layout()` does? Why is it explicitly written here?

Reply:
`libqasm` has a root folder with a `CMakeLists.txt` file and a `src` folder. I need `folders.source` to point to `.`, in order to read that top `CMakeLists.txt`, which is setting a variable that I need in `src/CMakeLists.txt`. I think I tried with `cmake_layout` and it didn't let me set `src_folder = "."`, but it was forcing me to set `src_folder = "src"`, hence this more verbose code.

Reply:
Looking at upstream CMakeLists, I don't think you need anything special, you can just use cmake_layout.

Reply:
Yes, this is what I had seen before (see my explanation above).

I have tried using `cmake_layout` again, and it has failed a [Conan v2 Lint](https://github.com/conan-io/conan-center-index/actions/runs/5119660631/jobs/9205202841?pr=17719).

It basically complains saying:
```
Error: recipes/libqasm/all/conanfile.py:53: [E9013(conan-layout-src-folder-is-src), LibqasmConan.layout] layout should set `src_folder` to `src`
```

But I cannot do that. I need to set my `src_folder` to `.`.

Reply:
> But I cannot do that. I need to set my src_folder to ..

Why?

Reply:
Because my main `CMakeLists.txt` is at `.` level. Then I have another `CMakeLists.txt` at `src` level.

Reply:
Using `cmake_layout` now.